architecture/deployment/oidc-deployment-and-configuration.md

```diff
@@ -0,0 +1,148 @@
+# Requirements
+
+In order for DataLabs to function there must be an OpenID Connection
+Authentication provider (e.g Auth0, Keycloak, AWS Cognito) which will either
+store user information directly or serve to re-direct users to a different
+authentication provider which in turn will allow a user to recieve an identity
+token.
+
+All providers are different and offer various advantages, there are multiple
+considerations to make including is how user registration is expected to work,
+and the convenience/limits of using a cloud (Auth0) vs a hosted solution.
+
+## Application Setup
+
+DataLabs requires an OIDC complient application to use for authentication. Hence
+once a provider is chosen an application must be set up for authentication, the
+method for doing this will vary depending on the provider, however some example
+guides can be found here;
+
+- [Auth0](https://auth0.com/docs/applications/set-up-an-application/register-single-page-app)
+- [Keycloak](https://robferguson.org/blog/2019/12/24/getting-started-with-keycloak/)
+
+Requirement for OIDC application;
+
+- The name should be clearly associated with "DataLabs"
+- The client must be OIDC compliant (this is typically a configuration option
+  during creation).
+- Callback URLs must be configured to match the callback URLs for the deployment
+  of DataLabs, this ensures that only requests which are re-directed back to the
+  application (and hence include secure credentials are not directed elsewhere)
+  are served. This should include the following URLS at minimum for;
```

Comment:
I'm finding this paragraph quite hard to follow. I think the bit brackets clashes slightly with the rest of the sentence and I think the use of `are served` makes me think that the OIDC provider is serving those pages. The following is my take on the same thing (and might highlight a complete lack of understanding on my part!)

---

Callback URLs must be configured to match the callback URLs of the DataLabs deployment. This ensures that the OIDC provider will only every redirect to your DataLabs deployment (after e.g. a user logging in) which is crucial as the redirects contain secure credentials. This should include the following URLs as a minimum: